CMakeLists.txt

```diff
@@ -134,15 +134,16 @@ if(OGS_USE_MPI)
     add_definitions(-DUSE_MPI)
 endif()
 
+add_definitions(-DEIGEN_INITIALIZE_MATRICES_BY_ZERO) # TODO check if needed
+if (EIGEN_NO_DEBUG)
+    add_definitions(-DEIGEN_NO_DEBUG)
```

Comment:
I would keep it as it can speed up debugging when you are sure that Eigen related parts are fine and there will be a bug in other places. Note that EIGEN_NO_DEBUG is OFF by default.